OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/fieldaccess/reflection/ReflectionRelatedFieldAccessesAnalysis.scala

```diff
@@ -0,0 +1,1055 @@
+            }
+
+        AllocationsUtil.continuationForAllocation[NameDepender, ContextType](
+            eps, state.callContext, data => (data.nameVar, data.stmts),
+            _.isInstanceOf[NameDepender], data => failure(data.pc, data.persistentActualParams, data.matchers)
+        ) { (data, _, allocationIndex, stmts) =>
+                val name = StringUtil.getString(allocationIndex, stmts)
+
+                val nameMatcher = retrieveSuitableMatcher[Set[String]](
+                    name.map(Set(_)),
+                    data.pc,
+                    v => new NameBasedFieldMatcher(v)
+                )
+
+                if (nameMatcher ne NoFieldsMatcher) {
+                    val matchers = data.matchers + nameMatcher
+                    val allMatchers = matchers +
+                        MatcherUtil.retrieveClassBasedFieldMatcher(
+                            data.accessContext,
+                            data.classVar,
+                            ClassDepender(data.pc, data.isStatic, data.persistentActualParams, matchers, data.classVar, data.stmts),
+                            data.pc,
+                            stmts,
+                            project,
+                            () => failure(data.pc, data.persistentActualParams, matchers),
+                            onlyFieldsExactlyInClass = false,
+                            considerSubclasses = !data.isStatic
+                        )
+
+                    addFieldAccesses(state.callContext, data.pc, allMatchers, data.persistentActualParams)
+                }
+            }
+
+        AllocationsUtil.continuationForAllocation[ClassDepender, ContextType](
+            eps, state.callContext, data => (data.classVar, data.stmts),
+            _.isInstanceOf[ClassDepender], data => failure(data.pc, data.persistentActualParams, data.matchers)
+        ) { (data, allocationContext, allocationIndex, stmts) =>
+                val classes = TypesUtil.getPossibleClasses(
+                    allocationContext, allocationIndex, data,
+                    stmts, () => failure(data.pc, data.persistentActualParams, data.matchers),
+                    onlyObjectTypes = false
+                ).flatMap { tpe =>
+                    if (data.isStatic) project.classHierarchy.allSubtypes(tpe.asObjectType, reflexive = true)
+                    else Set(if (tpe.isObjectType) tpe.asObjectType else ObjectType.Object)
+                }
+
+                val matchers = data.matchers +
+                    retrieveSuitableMatcher[Set[ObjectType]](
+                        Some(classes),
+                        data.pc,
+                        v => new ClassBasedFieldMatcher(v, false)
+                    )
+
+                addFieldAccesses(state.callContext, data.pc, matchers, data.persistentActualParams)
+            }
+
+        AllocationsUtil.continuationForAllocation[(ClassDepender, V, Array[Stmt[V]]), ContextType](
```

Comment:
What is this continuation for? Why does it use a tuple type instead of a case class like the others?

Reply:
See your other comment where you also asked which purpose this continuation has.